framework/src/meshgenerators/MeshRepairGenerator.C

```diff
@@ -61,112 +63,136 @@ MeshRepairGenerator::generate()
-              else if (_num_fixed_nodes == 10)
-                _console << "Node stitching will now proceed silently." << std::endl;
+              _console << "Two overlapping nodes in element " << elem->id() << " right by "
+                       << elem->vertex_average() << ".\n They will not be stitched" << std::endl;
+              continue;
             }
+
+            // Coordinates are the same but it's not the same node
+            // Replace the node in the element
+            const_cast<Elem *>(elem)->set_node(elem->get_node_index(&elem_node)) = node;
+            nodes_removed.insert(elem_node.id());
+
+            num_fixed_nodes++;
+            if (num_fixed_nodes < 10)
+              _console << "Stitching nodes " << *node << " and            " << elem_node
+                       << std::endl;
+            else if (num_fixed_nodes == 10)
+              _console << "Node stitching will now proceed silently." << std::endl;
           }
         }
       }
     }
-    _console << "Number of overlapping nodes which got merged: " << _num_fixed_nodes << std::endl;
-    if (mesh->allow_renumbering())
-      mesh->renumber_nodes_and_elements();
-    else
-    {
-      mesh->remove_orphaned_nodes();
-      mesh->update_parallel_id_counts();
-    }
   }
+  _console << "Number of overlapping nodes which got merged: " << num_fixed_nodes << std::endl;
+  if (mesh->allow_renumbering())
+  {
+    mesh->remove_orphaned_nodes();
+    mesh->renumber_nodes_and_elements();
+  }
+  else
+  {
+    mesh->remove_orphaned_nodes();
+    mesh->update_parallel_id_counts();
+  }
+}
 
-  // Flip orientation of elements to keep positive volumes
-  if (_fix_element_orientation)
-    MeshTools::Modification::orient_elements(*mesh);
-
-  // Disambiguate any block that has elements of multiple types
-  if (_elem_type_separation)
+void
+MeshRepairGenerator::separateSubdomainsByElementType(std::unique_ptr<MeshBase> & mesh) const
+{
+  std::set<subdomain_id_type> ids;
+  mesh->subdomain_ids(ids);
+  // loop on sub-domain
+  for (auto & id : ids)
   {
-    std::set<subdomain_id_type> ids;
-    mesh->subdomain_ids(ids);
-    // loop on sub-domain
-    for (auto & id : ids)
+    // Gather all the element types and blocks
+    // ElemType defines an enum for geometric element types
+    std::set<ElemType> types;
+    // loop on elements within this sub-domain
+    for (auto & elem : mesh->active_subdomain_elements_ptr_range(id))
+      types.insert(elem->type());
+
+    if (types.size() > 1)
     {
-      // Gather all the element types and blocks
-      // ElemType defines an enum for geometric element types
-      std::set<ElemType> types;
-      // loop on elements within this sub-domain
-      for (auto & elem : mesh->active_subdomain_elements_ptr_range(id))
-        types.insert(elem->type());
-
-      if (types.size() > 1)
+      auto next_block_id = MooseMeshUtils::getNextFreeSubdomainID(*mesh);
```

Comment:
`getNextFreeSubdomainID` will perform a global communication. However, `types` may not be globally consistent as the code is now and so it's possible that some processes will proceed to the global communication and others will not. Probably the thing to do is to communicate `types` before this `if` condition

Reply:
changed

Reply:
mesh is serialized btw, which for us currently means replicated.
Thanks for the idea though, this is future-proof now

Reply:
This would be a problem even with a serial mesh as you are determining `types` by looping over local elements

Reply:
I m using mesh->active_subdomain_elements_ptr_range(id))
not mesh->active_subdomain_(local)_elements_ptr_range(id)) (which does not exist)

and if I m reading this right, it calls active_subdomain_elements_begin ...
instead of active_local_subdomain_elements_begin (this one does exist)

Reply:
now is active_local_subdomain_elements_begin about local subdomains or local elements lol

no idea. Thank you for the fix

Reply:
> m using mesh->active_subdomain_elements_ptr_range(id))
> not mesh->active_subdomain_(local)_elements_ptr_range(id)) (which does not exist)

Ah I must have read it wrong. But glad we're in both a good current and future state